# One Sample Test (T-Test) (Instructional Worksheet)

## One Sample T-Test

**Goal:** Compare the mean value of a sample ($\mu$) with a constant value ($\mu_0$). 

We are interested in whether the following null or alternate hypotheses is supported by the data.

$$H_0: \mu = \mu_0$$
$$H_1: \mu \ne \mu_0$$

Because we are interested in comparing the population mean of one sample to a constant value, we will use a one sample t-test. 
Let's use the *ChickWeight* built-in dataset that we have used in previous modules. Just to recap, this dataset gives us the weight of different chicks that were fed 4 different diets.

In [1]:
import pandas as pd
from scipy import stats
ChickWeight = pd.read_csv('../data/ChickWeight.csv')

In [10]:
print(stats.ttest_1samp(ChickWeight.weight, 100))

import statsmodels.stats.api as sms

print(ChickWeight.weight.mean())
sms.DescrStatsW(ChickWeight.weight).tconfint_mean()

Ttest_1sampResult(statistic=7.380526033749192, pvalue=5.529345152599068e-13)
121.81833910034602


(116.01210691908608, 127.62457128160597)

The above code compares the mean weight of all chicks to a mean weight of 100 grams. If we look at the output, we are given the t-value and the p-value. From these results, we know the actual mean weight for all chicks is 121.8183, with a confidence interval of 116 to 128. Because the confidence interval does not cross the given constant mean (100), our results support the alternative hypothesis, the true mean is not equal to the constant given mean (100). We also know this because the p-value is significant (less than 0.05). 

In this example, we are running a 2-sided one sample t-test because we are interested in whether or not the mean of the sample is equal or not equal to the constant mean of 100. 

If we want a different percent confidence interval in our results, we can change the value in our *tconfint_mean()* function. Let's change it to 99%.

In [11]:
sms.DescrStatsW(ChickWeight.weight).tconfint_mean(.01)

(114.17839463591704, 129.458283564775)

## Problem Set

1. Using the *ChickWeight* dataset again, let's run a one sample t-test to see if the mean weight for chick 1 is equal or not equal to 100 grams. Remember that we are only interested in chick 1 now, so make sure you subset your dataset for this particular chick.

2. Is the mean weight for chick 1 significantly different than 100 grams? How do you know?

3. What is the 95% confidence interval? the 99% confidence interval?

4. Now, run a one sample t-test comparing the mean weight of chick 1 to a constant weight of 150 grams. What are your conclusions?

In [13]:
stats.ttest_1samp(ChickWeight[ChickWeight.Chick==1].weight, 100)

#chick 1 weight not significantly different than 100
#p value = 0.4984 (>0.05)
#confidence interval crosses 100
#fail to reject null hypothesis

Ttest_1sampResult(statistic=0.7000381849424331, pvalue=0.49844153563322424)

In [16]:
sms.DescrStatsW(ChickWeight[ChickWeight.Chick==1].weight).tconfint_mean(.05)
#95% confidence interval (74.99 - 148.35)

(74.98558161545712, 148.34775171787624)

In [17]:
sms.DescrStatsW(ChickWeight[ChickWeight.Chick==1].weight).tconfint_mean(.01)
#99% confidence interval (59.91 - 163.43)

(59.906048317122554, 163.4272850162108)

In [18]:
stats.ttest_1samp(ChickWeight[ChickWeight.Chick==1].weight, 150)

#chick 1 weight significantly differnt than 150 
#p value 0.042
#support alternative hypothesis - reject null hypothesis

Ttest_1sampResult(statistic=-2.3001254648108502, pvalue=0.04202265007409315)